### ISSUE TITLE
Read PNG Image constantly

### ISSUE THREAD

**Issue by netzfreak** [AndrewFromMelbourne/raspidmx#3]
Is it possible to a read PNG file continuously? For example every 2 seconds or check file size is different,  but without pngview will be closed and restart, without a short black screen.
As loop:
killall -S9 pngview; ./pngview alien.png &
Example so we can make png animation or read time and view it after convert in png.

Thanks,
Patrick

**Comment by fredericofs** [comment]
isnt what https://github.com/AndrewFromMelbourne/raspidmx/tree/master/spriteview does?

**Comment by AndrewFromMelbourne** [comment]
@netzfreak It certainly would be possible to do, but it isn't something that I have time to do at the moment.

@fredericofs The spriteview program displays different parts of a single PNG image. It only reads the image once. These are known as sprite sheets.

**Comment by joachimvenaas** [comment]
I also have this problem. Is there a way to open a second image over the first one, and then close the first one?

**Comment by AndrewFromMelbourne** [comment]
No, unfortunately there is no way to do that currently @joachimvenaa.

**Comment by joachimvenaas** [comment]
Found a solution that works with me

''''os.system(PNGVIEWPATH + "/pngview -b 0 -l 300003 -x 650 -y 5 " + ICONPATH + "/battery25.png &")
           
            out = check_output("ps aux | grep pngview | awk '{ print $2 }'", shell=True)
            nums = out.split('\n')
            for num in nums:
                i += 1
                if i == 1:
                    os.system("sudo kill " + killid)'''

This first displays the new png, then kills the old one

**Comment by TheRealFlyingCoder** [comment]
So I definitely love opening up a 4 year old issue 👍 

A very specific use case resulted in this being the perfect tool for me! But I too need an updating image layer.

I've got a while loop that can re-run code anytime the image file has been updated... but without any comments or instructions i'm struggling to figure out how to update that image layer so that it propagates

![image](https://user-images.githubusercontent.com/58131578/91409862-6459ed00-e889-11ea-9e75-df48d960688d.png)

Are any of the imagelayer.c helper functions suitable for this? Or would I have to destroy and recreate the image layer each time?

**Comment by magdesign** [comment]
@TheRealFlyingCoder : could you solve this somehow?
